app/script.rb

```diff
@@ -0,0 +1,184 @@
+        @accounts << account
+      end
+    end
+  end
+
+  def check_transactions
+    browser.goto(TRANSACTIONS_URL)
+    puts "Fetching transactions for the last two months"
+    @accounts.each do |account|
+      browser.div(class: "chosen-container").click
+      browser.div(class: "chosen-drop").span(text: "#{account.name}").click
+      sleep 2
+      set_date
+      sleep 2 
+      transactions_html.css('li.history-item.success').each do |page|
+        year  = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[1]
+        month = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[0]
+        day   = page.parent.parent.css('div.day-header').text.split[0]
+        time  = page.css('span.history-item-time').text
+        date  = (year + " " + month + " " + day).to_s
+        description = page.css('span.history-item-description').text.split.join(" ")
+        if !page.css('span.history-item-amount.transaction.income').text.empty?
+          amount = page.css('span.history-item-amount.transaction.income').text
+        elsif !page.css('span.history-item-amount.total').text.empty?
+          amount = page.css('span.history-item-amount.total').text
+        else
+          amount = page.css('span.history-item-amount.transaction').text
+        end
+        transaction = Transactions.new(date, description, amount)
+        account.transactions << transaction
+      end
+    end
+  end
+
+  def set_date
+    day = Date.today.prev_month(2).day.to_s
+    browser.input(name: 'from').click
+    browser.a(class: %w"ui-datepicker-prev ui-corner-all").click
+    browser.a(text: "#{day}").click
+  end
+
+  def assemble
+    hash = Hash.new
+    hash["accounts"] = []
+    @accounts.map do |account|
+      account_hash = Hash.new
+      account_hash['name'] = account.name
+      account_hash['balance'] = account.name
+      account_hash['currency'] = account.currency
+      account_hash['nature'] = account.nature
+      account_hash['transactions'] = []
+      account.transactions.map do |transaction|
+        transaction_hash = Hash.new
+        transaction_hash['date'] = transaction.date
+        transaction_hash['description'] = transaction.description
+        transaction_hash['amount'] = transaction.amount
+        account_hash['transactions'] << (transaction_hash)
+      end
+      hash["accounts"] << account_hash
+    end
+    hash
+  end
+
+  def store
+    Dir.mkdir('data') unless File.exists?('data')
+    file_name = 'data/accounts.json'
+    File.open(file_name, 'w') { |file| file.write(JSON.pretty_generate(assemble)) }
+    puts "Accounts saved to #{file_name}"
+  end
+end
+
+class Accounts
+  attr_accessor :name, :balance, :currency, :nature, :transactions
+
+  def initialize (name, currency, balance, nature)
+    @name         = name
+    @currency     = currency
+    @balance      = balance
+    @nature       = nature
+    @transactions = Array.new
```

Comment:
Please attach to 1 concept. If you create a new array with `Array.new` write it everywhere in such a way to be consistent. And vice-versa: if you create it with